docs/architecture/raft.rst

```diff
@@ -44,7 +44,12 @@ Enabling Raft
 
     See :doc:`the upgrade guide from 5.2 to 5.4 </upgrade/index>` for details.
 
-ScyllaDB Open Source 5.2 and later, and ScyllaDB Enterprise 2023.1 and later come equipped with a procedure that can setup Raft-based consistent cluster management in an existing cluster. We refer to this as the **Raft upgrade procedure** (do not confuse with the :doc:`ScyllaDB version upgrade procedure </upgrade/index/>`).
+.. warning::
+    Raft is mandatory in ScyllaDB Open Source 6.0 and later, and ScyllaDB Enterprise 2024.2 and later. You cannot prevent Raft from getting enabled upon upgrade to 6.0 or 2024.2.
+
+.. TODO: add reference to the upgrade guide from 5.4 to 6.0
+
+ScyllaDB Open Source 5.2 and 5.4, and ScyllaDB Enterprise 2023.x and 2024.1 come equipped with a procedure that can setup Raft-based consistent cluster management in an existing cluster. We refer to this as the **Raft upgrade procedure** (do not confuse with the :doc:`ScyllaDB version upgrade procedure </upgrade/index/>`).
```

Comment:
what was wrong with "and later"?

Reply:
My (maybe incorrect) logic was that this section describes enabling Raft without a version upgrade, and that isn't possible in versions later than 5.4. If you have a properly working cluster in 5.5 or later, then you must have already enabled Raft (during rolling upgrade to 5.5 at the latest). So, I thought that this "and later" might be confusing.

But since we are going to remove the "Enabling Raft" section (the discussion below), it doesn't matter anymore.

Reply:
All versions starting from 5.2 have the "Raft upgrade procedure".

It's true that eventually, this procedure will not be needed (if we forget about recovery for a while) because the version you're upgrading *from* already went through the procedure.

But 5.5 still needs the procedure, because you can have a cluster in 5.4 with Raft disabled, then you do a rolling upgrade. As soon as all nodes have upgraded to version 5.5, that's when the Raft upgrade procedure starts. It starts *after* rolling upgrade, not in the middle of it. For versions which still have the consistent_cluster_management flag -- the procedure starts *after* all nodes enable the flag.